src/databricks/labs/ucx/source_code/notebook_linter.py

```diff
@@ -0,0 +1,48 @@
+from collections.abc import Iterable
+
+from databricks.labs.ucx.source_code.base import Advice
+from databricks.labs.ucx.source_code.notebook import Notebook
+from databricks.labs.ucx.source_code import languages
+from databricks.labs.ucx.source_code.languages import Languages, Language
+
+
+class NotebookLinter:
+    """
+    Parses a Databricks notebook and then applies available linters
+    to the code cells according to the language of the cell.
+    """
+
+    def __init__(self, langs: Languages, notebook: Notebook):
+        self._languages: Languages = langs
+        self._notebook: Notebook = notebook
+        self._cell_offsets: list[int] = []
+
+    @classmethod
+    def from_notebook(cls, langs: Languages, notebook: Notebook) -> 'NotebookLinter':
+        return cls(langs, notebook)
+
+    @classmethod
+    def from_source(
+        cls, langs: Languages, source: str, default_language: Language = languages.Language.SQL
```

Comment:
```suggestion
        cls, langs: Languages, source: str, default_language: Language
```
So default language has a semantic meaning: "primary language for this given source code, while allowing for use of other languages wrapped up in magic comments". Having a default value for this argument would lead to wrong assumptions

Reply:
Got you